src/InteractiveSlide/InstantBook.php

```diff
@@ -147,97 +145,116 @@ function (CacheItemInterface $item) use ($configuration): mixed {
     }
 
     /**
-     * @throws \Throwable
+     * @throws InteractiveSlideException
+     * @throws InvalidArgumentException
      */
     private function getQuickBookOptions(Slide $slide, InteractionSlideRequest $interactionRequest): array
     {
         $resource = $interactionRequest->data['resource'] ?? null;
 
         if (null === $resource) {
-            throw new \Exception('Resource not set.');
+            throw new InteractiveSlideException('Resource not set.', 400);
         }
 
+        $start = (new \DateTime())->setTimezone(new \DateTimeZone('UTC'));
+        $startFormatted = $start->format('c');
+
         return $this->interactiveSlideCache->get(self::CACHE_KEY_OPTIONS_PREFIX.$resource,
-            function (CacheItemInterface $item) use ($slide, $resource, $interactionRequest) {
+            function (CacheItemInterface $item) use ($slide, $resource, $interactionRequest, $start, $startFormatted) {
                 $item->expiresAfter(new \DateInterval(self::CACHE_LIFETIME_QUICK_BOOK_OPTIONS));
 
-                /** @var User|ScreenUser $activeUser */
-                $activeUser = $this->security->getUser();
-                $tenant = $activeUser->getActiveTenant();
+                try {
+                    /** @var User|ScreenUser $activeUser */
+                    $activeUser = $this->security->getUser();
+                    $tenant = $activeUser->getActiveTenant();
```

Comment:
Domain service should not depend on the security context. Controllers extract relevant HTTP/Security context then pass to services. Tenant should be passed down as method parameter.